app/components/edit/EditAddress.jsx

```diff
@@ -1,129 +1,252 @@
-        onChange={handleFieldChange}
-      />
-      <input
-        type="text"
-        className="input"
-        placeholder="Country"
-        data-field="country"
-        value={address.country}
-        onChange={handleFieldChange}
-      />
-      <input
-        type="text"
-        className="input"
-        placeholder="Postal/Zip Code"
-        data-field="postal_code"
-        value={address.postal_code}
-        onChange={handleFieldChange}
-      />
-    </div>
+    <ReactModal
+      className={s.modal}
+      overlayClassName={s.overlay}
+      isOpen={isOpen}
+      onRequestClose={onRequestClose}
+    >
+      <form className={s.modalContent} onSubmit={handleSubmit}>
+        <h1 className={s.title}>{title}</h1>
+        <div className={s.formBody}>
+          <label className={s.inputLabel}>
+            Address name (optional)
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="name"
+              placeholder="Name"
+              defaultValue={defaultData.name}
+            />
+          </label>
+          <div className={s.formControlGroup}>
+            <span className={s.formControlGroupLabel}>Street Address</span>
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="address_1"
+              placeholder="Address"
+              required
+              defaultValue={defaultData.address_1}
+            />
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="address_2"
+              placeholder="Apartment, suite, unit, building, floor, etc."
+              defaultValue={defaultData.address_2}
+            />
+          </div>
+          <label className={s.inputLabel}>
+            City
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="city"
+              placeholder="City"
+              required
+              defaultValue={defaultData.city}
+            />
+          </label>
+          <label className={s.inputLabel}>
+            State
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="state_province"
+              placeholder="State"
+              required
+              defaultValue={defaultData.state_province}
+            />
+          </label>
+          <label className={s.inputLabel}>
+            Zip code
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="postal_code"
+              placeholder="Zip Code"
+              required
+              defaultValue={defaultData.postal_code}
+            />
+          </label>
+        </div>
+        <div className={s.actionRow}>
+          <button
+            className={s.cancelButton}
+            type="button"
+            onClick={onRequestClose}
+          >
+            ← Cancel
```

Comment:
`←` is probably better as an equivalent material icon here, I know the raw `←` character often renders weirdly off-center relative to the text on a the same line, its baseline is rarely perfectly centered.

Reply:
Yeah, good point. I swapped it out with a Remix icon, although it looks like our vertical alignment is still just a bit messed up in general due to a global `svg { vertical-align: middle }` CSS style that I'm not going to mess with right now. I would like to get rid of most of our global CSS at some point, since it's pretty dated and sometimes causes more issues with modern, idiomatic CSS.

<img width="141" alt="Screen Shot 2021-05-23 at 1 15 23 PM" src="https://user-images.githubusercontent.com/1002748/119275325-35c73380-bbc9-11eb-8d2d-3b644c974c7d.png">